src/main/java/com/hivemq/bootstrap/netty/initializer/AbstractChannelInitializer.java

```diff
@@ -73,45 +71,31 @@ protected void initChannel(@NotNull final Channel ch) throws Exception {
         ch.pipeline().addLast(MQTT_MESSAGE_DECODER, new MQTTMessageDecoder(channelDependencies));
         ch.pipeline().addLast(MQTT_MESSAGE_ENCODER, channelDependencies.getMqttMessageEncoder());
         addNoConnectIdleHandler(ch);
+        //MQTT_5_FLOW_CONTROL_HANDLER is added here after CONNECT
         ch.pipeline().addLast(MQTT_MESSAGE_BARRIER, new MessageBarrier(channelDependencies.getMqttServerDisconnector()));
         ch.pipeline().addLast(MQTT_SUBSCRIBE_MESSAGE_BARRIER, new SubscribeMessageBarrier());
         // before connack outbound interceptor as it initializes the client context after the connack
         ch.pipeline().addLast(PLUGIN_INITIALIZER_HANDLER, channelDependencies.getPluginInitializerHandler());
```

Comment:
PluginInitializerHandler is currently a ChannelDuplexHandler, but can be only a ChannelOutboundHandler
+ it should remove itself after `fireInitialize`